packages/ckeditor5-mention/src/mention.ts

```diff
@@ -32,10 +32,27 @@ export default class Mention extends Plugin {
 	 * // { id: '@joe', userId: '1234', uid: '7a7bc7...', _text: '@John Doe' }
 	 * ```
 	 *
-	 * @param viewElement
 	 * @param data Additional data to be stored in the mention attribute.
 	 */
-	public toMentionAttribute( viewElement: Element, data?: MentionAttribute ): MentionAttribute | undefined {
+	public toMentionAttribute<MentionData extends Record<string, unknown>>(
+		viewElement: Element,
+		data: MentionData
+	): ( MentionAttribute & MentionData ) | undefined;
+
+	/**
+	 * Creates a mention attribute value from the provided view element and optional data.
```

Comment:
For both signatures of the `toMentionAttribute` method, we should change descriptions, because both have the phrase "optional data". In the first one `data` parameter is mandatory and in the second one there is no `data` parameter.